client/basic-authorship-ver/src/basic_authorship.rs

```diff
@@ -400,24 +470,48 @@ where
 		let now = (self.now)();
 		let left = deadline.saturating_duration_since(now);
 		let left_micros: u64 = left.as_micros().saturated_into();
-		// NOTE reduce deadline by half as we want to avoid situation where
-		// fully filled previous block does not allow for any extrinsic to be included in following
-		// one
-		let soft_deadline = now +
-			time::Duration::from_micros(self.soft_deadline_percent.mul_floor(left_micros) / 2);
+		let first_slot_limit =
+			futures_timer::Delay::new(time::Duration::from_micros(left_micros * 55 / 100));
+
+		// let queue_processing_deadline = now + time::Duration::from_micros(left_micros / 2);
+		let queue_processing_deadline = now + time::Duration::from_micros(left_micros * 55 / 100);
+
 		let block_timer = time::Instant::now();
 		let mut skipped = 0;
 		let mut unqueue_invalid = Vec::new();
 
 		let mut t1 = self.transaction_pool.ready_at(self.parent_number).fuse();
-		// NOTE reduce deadline by half ('/16' instead of '/8') as we want to avoid situation where
-		// fully filled previous block does not allow for any extrinsic to be included in following
-		// one
-		let mut t2 =
-			futures_timer::Delay::new(deadline.saturating_duration_since((self.now)()) / 16).fuse();
 
-		block_builder.apply_previous_block_extrinsics(seed.clone());
+		let mut block_size = 0;
+
+		let get_current_time = &self.now;
+		let is_expired = || get_current_time() > queue_processing_deadline;
+
+		let block_size_limit = block_size_limit.unwrap_or(self.default_block_size_limit);
+		block_builder.apply_previous_block_extrinsics(
```

Comment:
fetching previous block txs up to half of the slot limit

Reply:
Why is it applying previous block extrinsics, when it should be executing them from queue?

Reply:
yeah, ill improve naming - thats exactly what its doing